### PR TITLE
Reclaim memory by putting constants in PROGMEM where possible

### PR DESCRIPTION
**Pull request by CurlyMoo**
Before:
Global variables use 52244 bytes (63%) of dynamic memory, leaving 29676 bytes for local variables. Maximum is 81920 bytes.

After:
Global variables use 51616 bytes (63%) of dynamic memory, leaving 39772 bytes for local variables. Maximum is 81920 bytes.

Thus saving 10096 bytes which can store quite an extensive ruleset :)

-----

This PR is the first of a series working towards implementing rules.

### CODE REVIEW

**Comment by CurlyMoo** [comment]
As with my previous PR. Don't just close it when not satisfied, but allow me to apply your feedback.

I've tested this commit on a local ESP8266 not connected to my heatpump. The webserver and such loads fine. So it seems this commit doesn't have any negative side effects.

File: HeishaMon/decode.cpp [hunk 1]
```diff
@@ -189,43 +191,37 @@ void decode_optional_heatpump_data(char* data, String actOptData[], PubSubClient
...
-        Topic_Name = "Alarm_State";
         break;
       default:
         break;
     }
     if ((updatenow) || ( actOptData[Topic_Number] != Topic_Value )) {
       actOptData[Topic_Number] = Topic_Value;
-      sprintf(log_msg, "received OPT%d %s: %s", Topic_Number, Topic_Name.c_str(), Topic_Value.c_str()); log_message(log_msg);
-      sprintf(mqtt_topic, "%s/%s/%s", mqtt_topic_base, mqtt_topic_pcbvalues, Topic_Name.c_str()); mqtt_client.publish(mqtt_topic, Topic_Value.c_str(), MQTT_RETAIN_VALUES);
+      sprintf_P(log_msg, PSTR("received OPT%d %s: %s"), Topic_Number, topics[Topic_Number], Topic_Value.c_str());
```
Comment (IgorYbema) [inline HeishaMon/decode.cpp:221]: shouldn't this be optTopics?

File: HeishaMon/decode.cpp [hunk 2]
```diff
@@ -189,43 +191,37 @@ void decode_optional_heatpump_data(char* data, String actOptData[], PubSubClient
...
+      log_message(log_msg);
+      sprintf(mqtt_topic, "%s/%s/%s", mqtt_topic_base, mqtt_topic_pcbvalues, topics[Topic_Number]);
```
Comment (IgorYbema) [inline HeishaMon/decode.cpp:223]: and here also optTopics?

**Review by IgorYbema — COMMENTED** [review]
Didn't knew soo much could be put in PROGMEM. Already did a lot though before :)

**Comment by CurlyMoo** [comment]
This PR was done against an older version of the HeishaMon code. Please wait for review when i ported it to the latest HeishaMon code.

File: HeishaMon/decode.cpp (hunk 1 above)
Comment (CurlyMoo) [reply HeishaMon/decode.cpp:221]: Yes

File: HeishaMon/decode.cpp (hunk 2 above)
Comment (CurlyMoo) [reply HeishaMon/decode.cpp:223]: Yes

**Review by IgorYbema — APPROVED** [review]

**Comment by CurlyMoo** [comment]
So, let me remove that bin file 👍

**Comment by CurlyMoo** [comment]
Done

**Pull request merged**